corehq/apps/hqwebapp/views.py

```diff
@@ -1095,6 +1096,18 @@ def deal_with_doc(doc, domain, doc_info_fn):
             domain = doc.domain
             return deal_with_doc(doc, domain, get_object_info)
 
+    for django_model in (SQLLocation,):
+        try:
+            if hasattr(django_model, 'by_id') and callable(django_model.by_id):
+                doc = django_model.by_id(query)
+            else:
+                doc = django_model.objects.get(pk=query)
+        except django_model.DoesNotExist:
+            pass
+        else:
+            domain = doc.domain
+            return deal_with_doc(doc, domain, get_object_info)
+
```

Comment:
This seems fine, but I'll note that we already have a similar interface for this kind of thing here: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/hqadmin/views.py#L684-L684
I commented on this issue on [this ticket](https://manage.dimagi.com/default.asp?258744#BugEvent.1368885), but I didn't see that there were two tickets about it :(